### DISCUSSION TITLE
Is their support for special characters in primary key? what if i need to have it?

### DISCUSSION THREAD

**Discussion by rz-vishalkumar** [meilisearch/product#456] [Feedback & Feature Proposal]

**Comment by gmourier** [comment]
Hi @rz-vishalkumar 👋 

Are you pointing out the value defining an identifier or the field name that you define as a `primaryKey`? 

Thanks!

**Comment by rz-vishalkumar** [comment]
I am asking in the value. There is a need in my project to use special
characters in primary key. :(

**Comment by gmourier** [comment]
Thanks @rz-vishalkumar! 

@Kerollmops Is there a particular constraint to not accept any UTF-8 character for the identifier of a document as long as it is a string? Thanks!

**Reply by Kerollmops** [reply]
The only reason I see is about URL-encoding behaviors and misusages, it can be a problem when you need to access and/or [delete a particular document by using its document id](https://docs.meilisearch.com/reference/api/documents.html#get-one-document) and encoding it in the URL. There are a lot of issues related to that in alternative-to-Meilisearch repositories.
